server/controllers/finance/debtors/index.js

```diff
@@ -64,28 +65,43 @@ function detail(req, res, next) {
 /**
  * Updates a debtor's details (particularly group_uuid)
  */
-function update(req, res, next) {
-  const sql =
-    'UPDATE debtor SET ? WHERE uuid = ?';
-
+async function update(req, res, next) {
+  const sql = `UPDATE debtor SET ? WHERE uuid = ?`;
+  const previousGroupSql = 'SELECT BUID(group_uuid) as group_uuid  FROM debtor WHERE uuid=?';
+  const historicSQL = `INSERT INTO debtor_group_history SET ?`;
+  const { user } = req.session;
+  const data = _.clone(req.body);
   // delete the uuid if it exists
-  delete req.body.uuid;
+  delete data.uuid;
 
   // cache the incoming uuid for fast lookups
   const uid = db.bid(req.params.uuid);
 
+  // let get some informations for historic
+  const previousGroup = await db.one(previousGroupSql, uid);
   // escape the group_uuid if it exists
   if (req.body.group_uuid) {
-    req.body.group_uuid = db.bid(req.body.group_uuid);
+    data.group_uuid = db.bid(req.body.group_uuid);
+  }
+  const transaction = db.transaction();
+  transaction.addQuery(sql, [data, uid]);
+
+  // check if we should update the historic table
+  if (req.body.group_uuid !== previousGroup.group_uuid) {
```

Comment:
Instead of just checking the previous debtor_group_uuid, can we just assume that if they sent the `group_uuid` they are changing the debtor group?  That would make this simpler.